packages/typography/src/styled/StyledCode.ts

```diff
@@ -6,17 +6,40 @@
  */
 
 import styled, { css, DefaultTheme, ThemeProps } from 'styled-components';
-import { DEFAULT_THEME, getColorV8, retrieveComponentStyles } from '@zendeskgarden/react-theming';
+import { DEFAULT_THEME, getColor, retrieveComponentStyles } from '@zendeskgarden/react-theming';
 import { StyledFont, IStyledFontProps } from './StyledFont';
 import { ICodeProps } from '../types';
 
 const COMPONENT_ID = 'typography.code';
 
-const colorStyles = (props: IStyledCodeProps & ThemeProps<DefaultTheme>) => {
-  const hue = props.hue || 'neutralHue';
-  const backgroundColor = getColorV8(hue, 200, props.theme);
-  const shade = hue === 'yellow' ? 800 : 700;
-  const foregroundColor = getColorV8(hue, shade, props.theme);
+const colorStyles = ({ hue, theme }: IStyledCodeProps & ThemeProps<DefaultTheme>) => {
+  const bgColorArgs: Parameters<typeof getColor>[0] = { theme };
+  const fgColorArgs: Parameters<typeof getColor>[0] = { theme };
+
+  switch (hue) {
+    case 'green':
+      bgColorArgs.variable = 'background.success';
+      fgColorArgs.variable = 'foreground.successEmphasis';
+      break;
+    case 'red':
+      bgColorArgs.variable = 'background.danger';
+      fgColorArgs.variable = 'foreground.dangerEmphasis';
+      break;
+    case 'yellow':
+      bgColorArgs.variable = 'background.warning';
+      fgColorArgs.variable = 'foreground.warningEmphasis';
+      break;
+    // includes grey
+    default:
+      fgColorArgs.variable = 'foreground.default';
+      bgColorArgs.hue = 'neutralHue';
+      bgColorArgs.dark = { shade: 900 };
+      bgColorArgs.light = { shade: 200 };
+      break;
```

Comment:
Unsupported hues default to grey. If a hue isn't recognized, `getColor` throws.

Reply:
This move unintentionally breaks the `Code` hue API. The expectation is that `hue` can receive any value that `getColor`'s hue argument can receive. So that includes all Garden hue values (lime, fuschia, etc), hex, etc. So a few things need to happen here:
- continue to give special treatment to Garden's primary colors
- get feedback from design on what shades to fallback to (perhaps dark:900 & light: 200)
- allow `getColor` to do it's thing – including throwing an error if someone provides a bogus value

Please see [StyledTag](https://github.com/zendeskgarden/react-components/blob/next/packages/tags/src/styled/StyledTag.ts) for comparison.

Reply:
`StyledCode` is "loosely" typed with `hue?: string`

https://github.com/zendeskgarden/react-components/blob/cffdb06680e0a63bd8faad7cc62085ec83b2f835/packages/typography/src/styled/StyledCode.ts#L31-L33

In our case, only `Code` uses this component and restricts possible `hue` values to only `grey`, `red`, `green`, and `yellow`.

https://github.com/zendeskgarden/react-components/blob/327ea52612f08bf0861d23853827cc2d8a590a7d/packages/typography/src/types/index.ts#L10

https://github.com/zendeskgarden/react-components/blob/327ea52612f08bf0861d23853827cc2d8a590a7d/packages/typography/src/types/index.ts#L84-L89

Therefore, I don't think it's a breaking change, but rather a call to improve clarity by redefining the `hue` type in `StyledCode` to match its real-life usage. I must emphasize this suggestion only make sense because `StyledCode` is exclusively used by `Code`.

```
interface IStyledCodeProps extends Omit<IStyledFontProps, 'size'> {
  hue?: (typeof HUE)[number];
  size?: ICodeProps['size'];
}
```
 
Toughts?

Reply:
Ah, interesting – I was looking at the wrong (`Styled`) API and testing on v8 with usage like `<Code hue="fuschia">`. In this case, restricting the `hue` color values makes sense. I still think allowing `getColor` to throw on invalid usage – as it now does in other v9 components – is appropriate and better than protecting the consumer from themselves.